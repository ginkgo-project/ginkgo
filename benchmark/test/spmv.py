#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output(
    ["-input", '[{"size": 100, "stencil": "7pt"}]'],
    expected_stdout="spmv.simple.stdout",
    expected_stderr="spmv.simple.stderr",
)

# stdin
test_framework.compare_output(
    [],
    expected_stdout="spmv.simple.stdout",
    expected_stderr="spmv.simple.stderr",
    stdin='[{"size": 100, "stencil": "7pt"}]',
)

# input file
test_framework.compare_output(
    ["-input", str(test_framework.sourcepath / "input.mtx.json")],
    expected_stdout="spmv.simple.stdout",
    expected_stderr="spmv.simple.stderr",
)

# profiler annotations
test_framework.compare_output(
    [
        "-input",
        '[{"size": 100, "stencil": "7pt"}]',
        "-profile",
        "-profiler_hook",
        "debug",
    ],
    expected_stdout="spmv.profile.stdout",
    expected_stderr="spmv.profile.stderr",
)
