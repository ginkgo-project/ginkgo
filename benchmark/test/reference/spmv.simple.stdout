
[
    {
        "size": 125,
        "stencil": "7pt",
        "spmv": {
            "coo": {
                "storage": 11600,
                "max_relative_norm2": 1.0,
                "time": 1.0,
                "repetitions": 10,
                "completed": true
            }
        },
        "nnz": 725,
        "optimal": {
            "spmv": "coo"
        }
    }
]
